Formula/libp11.rb

```diff
@@ -33,16 +33,27 @@ class Libp11 < Formula
 
   def install
     system "./bootstrap" if build.head?
-    system "./configure", *std_configure_args,
-                          "--disable-silent-rules",
-                          "--with-enginesdir=#{lib}/engines-1.1"
-    system "make", "install"
+    ["openssl@1.1", "openssl@3"].each do |pkg|
+      ver = pkg.split(/@/)[-1]
+      openssl = Formula[pkg]
+      next unless openssl.any_version_installed?
+
+      ENV.prepend_path "HOMEBREW_INCLUDE_PATHS", openssl.opt_include.to_s
+      ENV.prepend_path "HOMEBREW_LIBRARY_PATHS", openssl.opt_lib.to_s
+      ENV.prepend_path "PKG_CONFIG_PATH", "#{openssl.opt_lib}/pkgconfig"
+      system "./configure", *std_configure_args,
+             "--disable-silent-rules",
+             "--with-enginesdir=#{lib}/engines-#{ver}"
+      system "make"
+      system "make", "install"
+      system "make", "clean"
+    end
     pkgshare.install "examples/auth.c"
   end
 
   test do
```

Comment:
Can we test both here?

Reply:
@SMillerDev I'm not sure what tests you are saying. L47 is redundant and has been removed in an subsequent commit. The `make clean` intends to clean all stuff before compiling with another version of OpenSSL.
In order to test the pkcs11 engine libraries, I have added a little test in following commits to make sure those libraries related to a correct version of OpenSSL.